lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,11 +20,13 @@ def as_json
 
         def type_for(serializer)
           return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
+          type = serializer.object.class.to_s.split('::')
+          type.map! { |t| t.underscore.downcase }
```

Comment:
Ok, makes sense. How about if we leave simple transforms on strings in KeyTransform and move call on hash (with `deep_transform_keys!`) to where it's actually used/defined? Otherwise there'll be code repetition or misleading method naming... For example camel_lower is already doing that with hash keys, so I will have to come up with some not so clean name for simple method for strings.